src/stan/variational/advi.hpp

```diff
@@ -150,18 +150,143 @@ namespace stan {
       }
 
       /**
-       * Runs stochastic gradient ascent with Adagrad.
+       * Adaptively set hyperparameters for ADVI.
+       *
+       * @param variational variational distribution
+       * @param tuning_iter number of tuning iterations
+       * @return optimal value of eta in a coarse grid
+       */
+      double tune(Q& variational, int tuning_iter) const {
```

Comment:
can you add a unit test for this function? I think it's clear now that ADVI is solidly in the code base, so it should be treated as such.

In the test, I'm looking for
1. a way to instantiate the `tune` function easily.
2. that it actually tests the behavior. I think `variational` changes, so actually test that it changes to the right thing.
3. boundary conditions? perhaps `tuning_iter` at 0?